### ISSUE TITLE
Connected but not receiving data

### ISSUE THREAD

**Issue by seppelicous** [ottopaulsen/MMM-MQTT#23]
Hi,
I set up Magic Mirror on Docker and it works well. MQTT is also running in docker and works flawless with nodered. I can publish and receive topics in nodered.

So I installed this module and the log says, that it is connected and is subscribed to "strom" and "temp". I also see the labels in MM, but it does not show the payloads of both topics.
I published several messages with an iOS app, but nothing shows up in MM.

Is there something I could check again?

**Comment by seppelicous** [comment]
Okay, I checked the developer console in Safari and I found several JS errors:
Cant find variable topicsMatch
`                    if (sub.serverKey == payload.serverKey && topicsMatch(sub.topic, payload.topic)) {

`

and Invalid regular expression:
`const reg1 = /^\+$|^\+(?=\/)|(?<=\/)\+(?=\/)|(?<=\/)\+(?=$)/g;

`

**Comment by seppelicous** [comment]
So, today I installed MM on my RPI3 and unfortunately I am facing the same issues.
This is from the command line:

> Ready to go! Please point your browser to: http://localhost:8080
Create new calendar fetcher for url: http://www.calendarlabs.com/ical-calendar/ics/76/US_Holidays.ics - Interval: 300000
Create new news fetcher for url: http://www.nytimes.com/services/xml/rss/nyt/HomePage.xml - Interval: 300000
MMM-MQTT: Adding server:  { address: '192.168.178.5',
  port: '1883',
  user: '',
  password: '',
  subscriptions:
   [ { topic: 'strom',
       label: 'Zahl',
       sortOrder: 1,
       maxAgeSeconds: 600 },
     { topic: 'temp',
       label: 'Temperatur',
       suffix: '°C',
       decimals: 1,
       sortOrder: 10,
       maxAgeSeconds: 60 } ] }
MMM-MQTT: Starting client for:  { serverKey: '192.168.178.5:1883',
  address: '192.168.178.5',
  port: '1883',
  options: {},
  topics: [ 'strom', 'temp' ] }
MMM-MQTT: Connecting to mqtt://192.168.178.5:1883
MMM-MQTT connected to mqtt://192.168.178.5:1883
MMM-MQTT: subscribing to strom,temp



And this is from the developer tool in Safari:
<img width="1832" alt="Bildschirmfoto 2019-11-29 um 19 00 31" src="https://user-images.githubusercontent.com/34340436/69884861-a6dea100-12da-11ea-8668-b85564f14d66.png">

**Comment by ottopaulsen** [comment]
Hi

The regex that matches topics uses lookbehind, and this is pretty new in javascript, so you may need to upgrade node.js. What is your version? (Type `node --version` in the console to find out.)

Also, could you give an example of the exact messages you try to match, as they appear from MQTT? (Topics and values)

Otto

**Comment by seppelicous** [comment]
Hi Otto,
on Docker the version is 12.13.1 and on my raspberry I found a pretty old version: 10.17.0
I just updated nodejs on my Raspi to 13.2.0, but still got these errors in regex.

Problem with the docker image is, that I have to stick with the version the author provides on docker hub, which is 12.13.1.

I send the messages via nodered. This is an example of a message:

> {"topic":"strom","payload":"451","qos":0,"retain":false,"_msgid":"e56501c9.25e0c"}
